mmv1/products/datastream/api.yaml

```diff
@@ -369,3 +367,685 @@ objects:
             required: true
             description: |
               A free subnet for peering. (CIDR of /29)
+  - !ruby/object:Api::Resource
+    name: 'Stream'
+    base_url: "projects/{{project}}/locations/{{location}}/streams"
+    create_url: "projects/{{project}}/locations/{{location}}/streams?streamId={{stream_id}}"
+    self_link: "projects/{{project}}/locations/{{location}}/streams/{{stream_id}}"
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/datastream/docs/create-a-stream'
+      api: 'https://cloud.google.com/datastream/docs/reference/rest/v1/projects.locations.streams'
+    description: |
+      A resource representing streaming data from a source to a destination.
+    input: true
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: streamId
+        description: |-
+          The stream identifier.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'location'
+        description: |
+          The name of the location this stream is located in.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: The stream's name.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: Labels.
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        required: true
+        description: Display name.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'sourceConfig'
+        required: true
+        description: |
+          Source connection profile configuration.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'sourceConnectionProfile'
+            required: true
+            description: |
+              Source connection profile resource. Format: projects/{project}/locations/{location}/connectionProfiles/{name}
+          - !ruby/object:Api::Type::NestedObject
+            name: 'mysqlSourceConfig'
+            allow_empty_object: true
+            send_empty_value: true
+            exactly_one_of:
+              - source_config.0.mysql_source_config
```

Comment:
for all this file - why the name - `name: 'mysqlSourceConfig'` does not match the exactly_one_of ?

My understanding that this should be `source_config.0.mysql_source_config` only

Or - the name should be [source_stream_config](https://source.corp.google.com/piper///depot/google3/google/cloud/datastream/v1/datastream_resources.proto;bpv=1;bpt=1;l=805?q=datastream%20v1&gsn=source_stream_config&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dgoogle%2Fcloud%2Fdatastream%2Fv1%2Fdatastream_resources.proto%234.30.8.0)

Reply:
basically - the `name` attributes for these fields all match what's in the API, and are converted to snake case for Terraform. `exactly_one_of` and `conflicts` use the terraform field names. It's confusing because they also get specified in api.yaml, but that's how it is (for now at least.)

re: not using sourceStreamConfig - Terraform / YAML don't have the concept of "union" fields, so we have to model them as separate fields and use `exactly_one_of` and `conflicts` to indicate that only one can be specified. 

`conflicts` means that you can specify _at most one_. So this would I guess be a union field that is optional?
vs `exactly_one_of` which means that you _must_ specify one and only one. This would be a union field that is required.

Reply:
but shouldent a name = mysqlSourceConfig means that only source_config.0.mysql_source_config be in the exactly once?

Reply:
I don't understand the question, sorry - could you elaborate?